code/modules/paperwork/faxmachine.dm

```diff
@@ -80,130 +87,141 @@ GLOBAL_LIST_EMPTY(fax_blacklist)
 	else
 		to_chat(user, "<span class='warning'>You swipe the card through [src], but nothing happens.</span>")
 
-/obj/machinery/photocopier/faxmachine/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
-	if(!ui)
-		ui = new(user, src, ui_key, "faxmachine.tmpl", "Fax Machine UI", 540, 450)
-		ui.open()
-
-/obj/machinery/photocopier/faxmachine/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
-	var/data[0]
-	var/is_authenticated = is_authenticated(user)
-
-	if(scan)
-		data["scan_name"] = scan.name
-	else
-		data["scan_name"] = "-----"
-	data["authenticated"] = is_authenticated
-	if(!is_authenticated)
-		data["network"] = "Disconnected"
-	else if(!emagged)
-		data["network"] = fax_network
-	else
-		data["network"] = "ERR*?*%!*"
-	if(copyitem)
-		data["paper"] = copyitem.name
-		data["paperinserted"] = 1
-	else
-		data["paper"] = "-----"
-		data["paperinserted"] = 0
-	data["destination"] = destination
-	data["cooldown"] = sendcooldown
-	if((destination in GLOB.admin_departments) || (destination in GLOB.hidden_admin_departments))
-		data["respectcooldown"] = 1
-	else
-		data["respectcooldown"] = 0
-
-	return data
-
 /obj/machinery/photocopier/faxmachine/proc/is_authenticated(mob/user)
 	if(authenticated)
 		return TRUE
 	else if(user.can_admin_interact())
 		return TRUE
 	return FALSE
 
-/obj/machinery/photocopier/faxmachine/Topic(href, href_list)
-	if(..())
-		return 1
+/obj/machinery/photocopier/faxmachine/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "FaxMachine",  name, 540, 300, master_ui, state)
+		ui.open()
 
+/obj/machinery/photocopier/faxmachine/tgui_data(mob/user)
+	. = list()
+	.["authenticated"] = is_authenticated(user)
+	.["scan_name"] = scan ? scan.name : FALSE
+	if(!.["authenticated"])
+		.["network"] = "Disconnected"
+	else if(!emagged)
+		.["network"] = fax_network
+	else
+		.["network"] = "ERR*?*%!*"
+	.["paper"] = copyitem ? copyitem.name : FALSE
+	.["paperinserted"] = copyitem ? TRUE : FALSE
+	.["destination"] = destination ? destination : FALSE
+	.["sendError"] = FALSE
+	if(stat & (BROKEN|NOPOWER))
+		.["sendError"] = "No Power"
+	else if(!.["authenticated"])
+		.["sendError"] = "Not Logged In"
+	else if(!.["paper"])
+		.["sendError"] = "Nothing Inserted"
+	else if(!.["destination"])
+		.["sendError"] = "Destination Not Set"
+	else if((destination in GLOB.admin_departments) || (destination in GLOB.hidden_admin_departments))
+		var/cooldown_seconds = cooldown_seconds()
+		if(cooldown_seconds)
+			.["sendError"] = "Re-aligning in [cooldown_seconds] seconds..."
+
+
+/obj/machinery/photocopier/faxmachine/tgui_act(action, params)
+	. = ..()
+	if(.)
+		return
```

Comment:
```suggestion
	if(..())
		return
```
The other approach would return TRUE if someone had exploited the UI, which would update the UI even though it wasnt a valid action.